common/src/View/ScaleObjectsToolController.h

```diff
@@ -0,0 +1,74 @@
+#/*
+Copyright (C) 2010-2017 Kristian Duske
+
+This file is part of TrenchBroom.
+
+TrenchBroom is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+TrenchBroom is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with TrenchBroom. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#ifndef TrenchBroom_ScaleObjectsToolController
+#define TrenchBroom_ScaleObjectsToolController
+
+#include "Model/Hit.h"
+#include "Renderer/EdgeRenderer.h"
+#include "View/ToolController.h"
+
+namespace TrenchBroom {
+    namespace Model {
+        class PickResult;
+    }
+    
+    namespace Renderer {
+        class RenderBatch;
+        class RenderContext;
+    }
+    
+    namespace View {
+        class InputState;
+        class ScaleObjectsTool;
+        
+        class ScaleObjectsToolController : public ToolControllerBase<PickingPolicy, KeyPolicy, MousePolicy, MouseDragPolicy, RenderPolicy, NoDropPolicy> {
```

Comment:
We already discussed that this may benefit from one of the more specialized tool superclasses that support a dragsnapper / restricter. Maybe I'll see if I can rework it later on, so you don't have to change it now.